src/extensions/CanvasWidget/CanvasWidget.tsx

```diff
@@ -77,14 +77,15 @@ const dragReducer = (state: DragState, action: DragAction): DragState => {
     throw Error('Unexpected: unreachable, but keeps ESLint happy')
 }
 
-interface Props<T extends BaseLayerProps> {
-    layers: CanvasWidgetLayer<T, any>[] | null, // the layers to paint (each corresponds to a canvas html element)
+interface Props {
+    layers: (CanvasWidgetLayer<any, any> | null)[], // the layers to paint (each corresponds to a canvas html element)
+    width: number,
```

Comment:
But the canvas widget still needs to know the width/height... so those are passed to both the CanvasWidget props and to the individual layer props.